src/ace/managers/viewport/tilebuffer.c

```diff
@@ -658,24 +681,59 @@ void tileBufferRedrawAll(tTileBufferManager *pManager) {
 	PLANEPTR pDstPlane = pManager->pScroll->pBack->Planes[0];
 	tTileBufferTileIndex **pTileData = pManager->pTileData;
 	UWORD uwBltsize = tileBufferSetupTileDraw(pManager);
+#if defined ACE_DEBUG
+	if ((uwBltsize & BLIT_WORDS_NON_INTERLEAVED_BIT) && isInterleaved) {
+		logWrite("ERR: you called tileBufferRedrawAll%s, but the buffers are %s",
+			isInterleaved ? "Interleaved" : "NonInterleaved",
+			isInterleaved ? "not interleaved" : "interleaved"
+		);
+	}
+#endif
 	UWORD uwTileOffsX = (wStartX << ubTileShift);
-	UWORD uwDstOffsStep = ubTileSize / 8;
+	UWORD uwMarginedHeight = pManager->uwMarginedHeight;
+	UWORD uwWrapAroundY = MIN(uwEndY, uwMarginedHeight >> ubTileShift);
 	systemSetDmaBit(DMAB_BLITHOG, 1);
-	for (UWORD uwTileY = wStartY; uwTileY < uwEndY; ++uwTileY) {
-		UWORD uwTileCurr = wStartX;
-		ULONG ulDstOffs = uwDstBytesPerRow * uwTileOffsY + uwTileOffsX / 8;
-		while(uwTileCurr < uwEndX) {
+	systemGetBlitterFromOs();
```

Comment:
This isn't needed since ACE defaults to having blitter ownership - those functions are only used for file IO to make sure that OS has access to it, because it needs it in some cases.

I should've added comments explaining that for those functions.

If you're paranoid, you can add a check using `systemBlitterIsUsed()` if ACE has blitter ownership at this point and if not, print an `ERR:`.

Reply:
Changed the contents :)